### DISCUSSION TITLE
How to satisfy relay connection spec with prisma plugin and explicit many-to-many relations?

### DISCUSSION THREAD

**Discussion by villesau** [hayes/pothos#655] [Q&A] [answered]
Hi!

I have roughly a following structure: `Company` -> `CompanyUser(comanyId, userId, related info)` -> `User`. Ideally it would transform to roughly following graphql schema:

```
type User {
  id: ID!
  name: String!
}

type CompanyUsersConnection {
  edges: [CompanyUsersConnectionEdge!]!
  pageInfo: PageInfo!
}

type CompanyUsersConnectionEdge { // <-- this would expose what the underlying CompanyUser object exposes
  cursor: ID!
  node: User!
  companyRole: String! // <-- such as your role in the company and other properties, the info on how the user relates to company
}

type Company {
  id: ID!
  users(after: ID, before: ID, first: Int, last: Int): CompanyUsersConnection!
}
```

However, currently with following approach: `users: t.relatedConnection("users", { cursor: "userId_companyId" })`

I roughly end up with following schema:

```
type User {
  id: ID!
  name: String!
}

type CompanyProjectsConnection {
  edges: [CompanyProjectsConnectionEdge!]!
  pageInfo: PageInfo!
}

type CompanyUser {
  company: Company!
  id: ID!
  user: User!
}

type CompanyUsersConnection {
  edges: [CompanyUsersConnectionEdge!]!
  pageInfo: PageInfo!
}

type CompanyUsersConnectionEdge {
  cursor: ID!
  node: CompanyUser!
}

type Company {
  id: ID!
  users(after: ID, before: ID, first: Int, last: Int): CompanyUsersConnection!
}

```

So there is an unnecessary level of extra nesting. Ideally the edge would contain the CompanyUser info rather than the node. Is there a way  to achieve the first schema with `relatedConnection` somehow? Something such as

```
users: t.relatedConnection({
edgeField: "users", 
nodeField: "user",
cursor: "userId_companyId" })
```

**Comment by hayes** [comment]
this is something that hasn't been easy to do so far.  I am working on making it easier to build custom connections that still take advantage of the optimizations the prisma plugin can do.

Work in progress is in #671

**Comment by hayes** [comment, marked as answer]
Added some new docs on a potential way to implement this here: https://pothos-graphql.dev/docs/plugins/prisma#extending-connection-edges
